#VRML_SIM R2019b utf8
WorldInfo {
}
Viewpoint {
  orientation -0.28905613583909695 0.9187249096782993 0.26905592480071927 1.5609673560240371
  position 3.924683899756558 1.8086787997236629 0.1880032590223731
  follow "two_wheel_balance"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Solid {
  translation -0.5 0.05 1.21
  children [
    Shape {
      appearance Appearance {
      }
      geometry Box {
        size 0.1 0.2 0.1
      }
    }
  ]
  name "solid(6)"
  boundingObject Shape {
    appearance Appearance {
    }
    geometry Box {
      size 0.1 0.2 0.1
    }
  }
  physics Physics {
  }
}
Solid {
  translation 1.18 0.05 1.32
  children [
    Shape {
      appearance Appearance {
      }
      geometry Box {
        size 0.1 0.2 0.1
      }
    }
  ]
  name "solid(5)"
  boundingObject Shape {
    appearance Appearance {
    }
    geometry Box {
      size 0.1 0.2 0.1
    }
  }
  physics Physics {
  }
}
Solid {
  translation 1.33 0.05 -0.26
  children [
    Shape {
      appearance Appearance {
      }
      geometry Box {
        size 0.1 0.2 0.1
      }
    }
  ]
  name "solid(1)"
  boundingObject Shape {
    appearance Appearance {
    }
    geometry Box {
      size 0.1 0.2 0.1
    }
  }
  physics Physics {
  }
}
TexturedBackgroundLight {
}
UR5e {
  translation 0.25 -1.46724e-08 -1.02
  controller "my_controller004_1"
}
DEF two_wheel_balance Robot {
  translation 1.64 0.08692 -2.43798e-09
  rotation -1 0 0 2.7858271149692418e-08
  children [
    DEF inertial_unit InertialUnit {
      rotation 0 1 0 3.141592653589793
      name "inertial_unit"
      xAxis FALSE
      yAxis FALSE
    }
    DEF ds6 Transform {
      translation 0 0.025 0
      rotation 0 1 0 -2.2253003061004253
      children [
        DEF ds6 DistanceSensor {
          translation 0.05 0.005 0
          name "ds6"
          lookupTable [
            0 0 0
            1 1000 0
          ]
        }
      ]
    }
    DEF ds5 Transform {
      translation 0 0.025 0
      rotation 0 1 0 -2.4870996938995744
      children [
        DEF ds5 DistanceSensor {
          translation 0.05 0.005 0
          name "ds5"
          lookupTable [
            0 0 0
            1 1000 0
          ]
        }
      ]
    }
    DEF ds4 Transform {
      translation 0 0.025 0
      rotation 0 1 0 -2.8797996938995745
      children [
        DEF ds4 DistanceSensor {
          translation 0.05 0.005 0
          name "ds4"
          lookupTable [
            0 0 0
            1 1000 0
          ]
        }
      ]
    }
    DEF ds3 Transform {
      translation 0 0.025 0
      rotation 0 1 0 2.2253003061004253
      children [
        DEF ds3 DistanceSensor {
          translation 0.05 0.005 0
          name "ds3"
          lookupTable [
            0 0 0
            1 1000 0
          ]
        }
      ]
    }
    DEF ds2 Transform {
      translation 0 0.025 0
      rotation 0 1 0 2.4871003061004253
      children [
        DEF ds2 DistanceSensor {
          translation 0.05 0.005 0
          name "ds2"
          lookupTable [
            0 0 0
            1 1000 0
          ]
        }
      ]
    }
    DEF ds1 Transform {
      translation 0 0.025 0
      rotation 0 1 0 2.8797996938995745
      children [
        DEF ds1 DistanceSensor {
          translation 0.05 0.005 0
          name "ds1"
          lookupTable [
            0 0 0
            1 1000 0
          ]
        }
      ]
    }
    DEF led3 LED {
      translation 0.025 0 -0.025
      name "led3"
      color [
        0 1 0
      ]
    }
    DEF led2 LED {
      translation 0.025 0 0
      name "led2"
      color [
        1 1 1
      ]
    }
    DEF led1 LED {
      translation 0.025 0 0.025
      name "led1"
      color [
        1 1 0
      ]
    }
    DEF right_motor Transform {
      translation 0 -0.0375 -0.0875
      children [
        DEF right_motor Servo {
          translation 0 0 -0.025
          rotation 0 0 1 0
          children [
            DEF right_wheel Solid {
              translation 0 0 -0.0125
              rotation 1 0 0 1.5707996938995747
              children [
                DEF right_wheel Shape {
                  appearance Appearance {
                    material Material {
                    }
                    texture ImageTexture {
                      url [
                        "textures/asphalt.jpg"
                      ]
                    }
                  }
                  geometry Cylinder {
                    height 0.025
                    radius 0.05
                  }
                }
              ]
              name "right_wheel"
              boundingObject DEF right_wheel Shape {
                appearance Appearance {
                  material Material {
                  }
                  texture ImageTexture {
                    url [
                      "textures/asphalt.jpg"
                    ]
                  }
                }
                geometry Cylinder {
                  height 0.025
                  radius 0.05
                }
              }
              physics Physics {
              }
            }
          ]
          name "right_motor"
          boundingObject DEF right_motor Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0 0 0
              }
            }
            geometry Box {
              size 0.025 0.025 0.05
            }
          }
          physics Physics {
          }
          maxVelocity 1000
          position 2268.799998885379
        }
        DEF right_motor Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0 0 0
            }
          }
          geometry Box {
            size 0.025 0.025 0.05
          }
        }
      ]
    }
    DEF left_motor Transform {
      translation 0 -0.0375 0.0875
      children [
        DEF left_motor Servo {
          translation 0 0 0.025
          rotation 0 0 1 0
          children [
            DEF left_wheel Solid {
              translation 0 0 0.0125
              rotation 1 0 0 1.5707996938995747
              children [
                DEF left_wheel Shape {
                  appearance Appearance {
                    material Material {
                    }
                    texture ImageTexture {
                      url [
                        "textures/asphalt.jpg"
                      ]
                    }
                  }
                  geometry Cylinder {
                    height 0.025
                    radius 0.05
                  }
                }
              ]
              name "left_wheel"
              boundingObject DEF left_wheel Shape {
                appearance Appearance {
                  material Material {
                  }
                  texture ImageTexture {
                    url [
                      "textures/asphalt.jpg"
                    ]
                  }
                }
                geometry Cylinder {
                  height 0.025
                  radius 0.05
                }
              }
              physics Physics {
              }
            }
          ]
          name "left_motor"
          boundingObject DEF left_motor Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0 0 0
              }
            }
            geometry Box {
              size 0.025 0.025 0.05
            }
          }
          physics Physics {
          }
          maxVelocity 1000
          position 2269.744483657004
        }
        DEF left_motor Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0 0 0
            }
          }
          geometry Box {
            size 0.025 0.025 0.05
          }
        }
      ]
    }
    DEF body Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0.5 0 0
        }
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
  ]
  name "two_wheel_balance"
  boundingObject DEF body Shape {
    appearance Appearance {
      material Material {
        diffuseColor 0.561 0.561 0.561
      }
    }
    geometry Box {
      size 0.1 0.05 0.2
    }
  }
  physics Physics {
  }
  controller "my_controller04_2"
}
Pedestrian {
  translation 1.6 1.27 -1.7
}
Solid {
  translation -0.5 0.05 0.1
  children [
    Shape {
      appearance Appearance {
      }
      geometry Box {
        size 0.1 0.2 0.1
      }
    }
  ]
  boundingObject Shape {
    appearance Appearance {
    }
    geometry Box {
      size 0.1 0.2 0.1
    }
  }
  physics Physics {
  }
}
Solid {
  translation -0.5 0.05 -1.55
  children [
    Shape {
      appearance Appearance {
      }
      geometry Box {
        size 0.1 0.2 0.1
      }
    }
  ]
  name "solid(2)"
  boundingObject Shape {
    appearance Appearance {
    }
    geometry Box {
      size 0.1 0.2 0.1
    }
  }
  physics Physics {
  }
}
Solid {
  translation -1.3 0.05 -0.09
  children [
    Shape {
      appearance Appearance {
      }
      geometry Box {
        size 0.1 0.2 0.1
      }
    }
  ]
  name "solid(4)"
  boundingObject Shape {
    appearance Appearance {
    }
    geometry Box {
      size 0.1 0.2 0.1
    }
  }
  physics Physics {
  }
}
Solid {
  translation -1.63 0.05 0.76
  children [
    Shape {
      appearance Appearance {
      }
      geometry Box {
        size 0.1 0.2 0.1
      }
    }
  ]
  name "solid(3)"
  boundingObject Shape {
    appearance Appearance {
    }
    geometry Box {
      size 0.1 0.2 0.1
    }
  }
  physics Physics {
  }
}
CircleArena {
  radius 3
}
